### DISCUSSION TITLE
LOAD_DEFAULT_CONFIG_DATA

### DISCUSSION THREAD

**Discussion by hmueller01** [khoih-prog/ESPAsync_WiFiManager_Lite#19] [Ideas]
Why is LOAD_DEFAULT_CONFIG_DATA a `bool` and not a `#define`?
Is it needed to switch the behavior during runtime?
If this is a define some code and the variable `ESP_WM_LITE_Configuration defaultConfig;` could be removed in production code!

**Comment by khoih-prog** [comment]
Hi @hmueller01 

I'm happy that somebody having time to look at the library's source code. The library will certainly has better functions and much less bugs when having many more pairs of eyes.

There could have some real reason I use that `bool` way. Sorry I don't have time to look at / remember why I did that way.

You can try and figure out.

**Reply by hmueller01** [reply]
I tried and I didn't find a reason. So this is why I was asking. I will make a PR so we can test it.

**Reply by khoih-prog** [reply]
Anyway, be careful to test all the use-cases, because the var is all capital `LOAD_DEFAULT_CONFIG_DATA`, meaning I intended to use as in `#define`. But for some reason it didn't work correctly, and was changed to `bool`

I now don't think it's good to change, unless you can prove beyond any doubt the benefit of the modification. I also don't have time now to spend on every thing.
